--- build_detect_platform.orig
+++ build_detect_platform
@@ -73,7 +73,10 @@
         COMMON_FLAGS="$MEMCMP_FLAG -DOS_MACOSX"
         PLATFORM_SHARED_EXT=dylib
         [ -z "$INSTALL_PATH" ] && INSTALL_PATH=`pwd`
-        PLATFORM_SHARED_LDFLAGS="-dynamiclib -install_name $INSTALL_PATH/"
+        PLATFORM_SHARED_LDFLAGS="-dynamiclib"
+        PLATFORM_SHARED_LDFLAGS="$PLATFORM_SHARED_LDFLAGS -current_version $SHARED_MAJOR.$SHARED_MINOR.$SHARED_TINY"
+        PLATFORM_SHARED_LDFLAGS="$PLATFORM_SHARED_LDFLAGS -compatibility_version $SHARED_MAJOR.0.0"
+        PLATFORM_SHARED_LDFLAGS="$PLATFORM_SHARED_LDFLAGS -install_name $INSTALL_PATH/"
         PORT_FILE=port/port_posix.cc
         ;;
     Linux)
