--- libdecodeqr/Makefile.in.orig	2007-02-22 08:22:03.000000000 +0900
+++ libdecodeqr/Makefile.in	2012-12-24 02:01:31.000000000 +0900
@@ -21,8 +21,8 @@
 
 BASENAME=libdecodeqr
 LIBNAME_A=$(BASENAME).a
-SONAME=$(BASENAME).so.$(MAJOR)
-LIBNAME_SO=$(BASENAME).so.$(MAJOR).$(MINOR).$(TEENY)
+SONAME=$(BASENAME).$(MAJOR).dylib
+LIBNAME_SO=$(BASENAME).$(MAJOR).$(MINOR).$(TEENY).dylib
 
 CXX=@CXX@
 CPPFLAGS=@CPPFLAGS@ @CXXFLAGS@ -fPIC -c
@@ -30,10 +30,10 @@
 AR=@AR@
 ARFLAGS=rcs
 
-LD=@LD@
-LDFLAGS=@LDFLAGS@ -shared -soname $(SONAME)
+LD=@CXX@
+LDFLAGS=@LDFLAGS@ -dynamiclib -Wl,-headerpad_max_install_names,-undefined,dynamic_lookup,-compatibility_version,$(MAJOR).$(MINOR).0,-current_version,$(MAJOR).$(MINOR).$(TEENY),-install_name,$(libdir)/$(LIBNAME_SO)
 LIBCV=@LIBCV@
-LIBS=$(LIBCV)
+LIBS=-lopencv_core -lopencv_highgui -lopencv_imgproc -lc -lstdc++
 
 INSTALL=@INSTALL@
 LDCONFIG=/sbin/ldconfig
@@ -55,11 +55,5 @@
 $(LIBNAME_SO): $(objs)
 	$(LD) -o $@ $(objs) $(LDFLAGS) $(LIBS)
 
-install: $(dsts)
-	$(INSTALL) -m 0755 $(LIBNAME_A) $(LIBNAME_SO) $(libdir)
-	$(LDCONFIG) -n $(libdir)
-	$(LN) -sf $(libdir)/$(LIBNAME_SO) $(libdir)/$(BASENAME).so
-	$(INSTALL) -m 0644 decodeqr.h qrerror.h qrtypes.h $(includedir)
-
 clean:
-	-rm *.a *.o *.so.*
+	-rm *.a *.o *.so.* *.dylib
